src/bootstrap/bin/rustc.rs

```diff
@@ -169,8 +171,19 @@ fn main() {
             let is_test = args.iter().any(|a| a == "--test");
             // If the user requested resource usage data, then
             // include that in addition to the timing output.
-            let rusage_data =
-                env::var_os("RUSTC_PRINT_STEP_RUSAGE").and_then(|_| format_rusage_data());
+            let rusage_data = env::var_os("RUSTC_PRINT_STEP_RUSAGE").and_then(|_| {
```

Comment:
I would prefer if you structure this so that the `format_rusage_data` function takes the child as a formal parameter in both `cfg` cases.

Then the `cfg(not(windows))` version would just ignore its argument, and the `cfg(windows)` version would convert it to a handle.

This change would reduce the code here to a much simpler: 

```rust
env::var_os("RUSTC_PRINT_STEP_RUSAGE").and_then(|_| format_rusage_data(child));
```

How does that sound to you, @rylev ?